src/System.Management.Automation/engine/ExperimentalFeature/ExperimentalFeature.cs

```diff
@@ -0,0 +1,311 @@
+            return false;
+        }
+
+        /// <summary>
+        /// Log an error message.
+        /// </summary>
+        private static void LogError(PSEventId eventId, params object[] args)
+        {
+            PSEtwLog.LogOperationalError(eventId, PSOpcode.Constructor, PSTask.ExperimentalFeature, PSKeyword.UseAlwaysOperational, args);
+        }
+
+        /// <summary>
+        /// Check if the name follows the engine experimental feature name convention.
+        /// Convention: prefix 'PS' to the feature name -- PSFeatureName
+        /// </summary>
+        internal static bool IsEngineFeatureName(string featureName)
+        {
+            return featureName.IndexOf('.') == -1 && featureName.Length > 2 && featureName.StartsWith("PS", StringComparison.Ordinal);
+        }
+
+        /// <summary>
+        /// Check if the name follows the module experimental feature name convention.
+        /// Convention: ModuleName.FeatureName
+        /// </summary>
+        /// <param name="featureName">The feature name to check.</param>
+        /// <param name="moduleName">When specified, we check if the feature name matches the module name</param>
+        internal static bool IsModuleFeatureName(string featureName, string moduleName = null)
+        {
+            int firstDotIndex = featureName.IndexOf('.');
+            int lastDotIndex = featureName.LastIndexOf('.');
+
+            bool legit = firstDotIndex > 0 && lastDotIndex < featureName.Length - 1;
+            if (legit && moduleName != null)
+            {
+                var moduleNamePart = featureName.AsSpan(0, lastDotIndex);
+                return moduleNamePart.Equals(moduleName.AsSpan(), StringComparison.OrdinalIgnoreCase);
+            }
+            return legit;
+        }
+
+        /// <summary>
+        /// Determine the action to take for the specified experiment name and action.
+        /// </summary>
+        internal static ExperimentAction GetActionToTake(string experimentName, ExperimentAction experimentAction)
+        {
+            if (experimentName == null || experimentAction == ExperimentAction.None)
+            {
+                // If either the experiment name or action is not defined, then return 'Show' by default.
+                // This could happen to 'ParameterAttribute' when no experimental related field is declared.
+                return ExperimentAction.Show;
+            }
+
+            ExperimentAction action = experimentAction;
+            if (!HasEnabled(experimentName))
+            {
+                action = (action == ExperimentAction.Hide) ? ExperimentAction.Show : ExperimentAction.Hide;
+            }
+            return action;
+        }
+
+        /// <summary>
+        /// Check if the specified experimental feature has been enabled.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool HasEnabled(string featureName)
```

Comment:
`HasEnabled` seems odd here.  Do we use this naming elsewhere?  What about `Enabled` or `IsEnabled`

Reply:
Will change to `IsEnabled`